include/oneapi/dpl/pstl/tuple_impl.h

```diff
@@ -289,6 +306,36 @@ struct __copy_assignable_holder<_Tp, false> : oneapi::dpl::__internal::__value_h
     operator=(__copy_assignable_holder&& other) = default;
 };
 
+template <typename _Tuple1, typename _Tuple2, int I = 0>
+constexpr bool
+__equal(const _Tuple1& __lhs, const _Tuple2& __rhs)
+{
+    if constexpr (I < std::tuple_size_v<_Tuple1>)
+    {
+        return std::get<I>(__lhs) == std::get<I>(__rhs) &&
+               oneapi::dpl::__internal::__equal<_Tuple1, _Tuple2, I + 1>(__lhs, __rhs);
+    }
+    else
+    {
+        return true;
+    }
+}
+template <typename _Tuple1, typename _Tuple2, int I = 0>
+constexpr bool
+__less(const _Tuple1& __lhs, const _Tuple2& __rhs)
+{
+    if constexpr (I < std::tuple_size_v<_Tuple1>)
+    {
+        return std::get<I>(__lhs) < std::get<I>(__rhs) ||
```

Comment:
Is it an error?
I mean that if `std::get<I>(__lhs) > std::get<I>(__rhs)` then we shouldn't call anything else, I think:
https://en.cppreference.com/w/cpp/utility/tuple/operator_cmp 
```C++
template< class... TTypes, class... UTypes >
constexpr bool operator<( const [std::tuple](http://en.cppreference.com/w/cpp/utility/tuple)<TTypes...>& lhs,
                          const [std::tuple](http://en.cppreference.com/w/cpp/utility/tuple)<UTypes...>& rhs );
(3)
`true` if the first non-equivalent element in `lhs` is less than the one in `rhs`, `false` if the first non-equivalent element in `rhs` is less than the one in `lhs` or there is no non-equivalent element. For two empty tuples, returns `false`.
```